patches/tModLoader/Terraria/Projectile.TML.cs

```diff
@@ -86,6 +88,37 @@ private static void HandlePlayerStatModifiers(IEntitySource spawnSource, Project
 		}
 	}
 
+	/// <summary>
+	/// Will drop loot the same way as when <see cref="ProjectileID.Geode"/> is cracked open.
+	/// </summary>
+	/// <param name="entity">The entity the loot originates from</param>
+	public static void DropGeodeLoot(Entity entity)
+	{
+		var dict = ItemID.Sets.GeodeDrops;
+		foreach (var pair in dict) {
+			string exceptionCommon = $"{Lang.GetItemNameValue(pair.Key)} registered in 'ItemID.Sets.{nameof(ItemID.Sets.GeodeDrops)}'";
+			if (pair.Value.minStack < 1)
+				throw new Exception($"{exceptionCommon} must have minStack bigger than 0");
+
+			if (pair.Value.maxStack <= pair.Value.minStack)
+				throw new Exception($"{exceptionCommon} must have maxStack bigger than minStack");
+		}
+		var list = dict.Keys.ToList();
+
+		int attempts = 0;
+		while (attempts < 2 && list.Count > 0) {
+			attempts++;
+
+			int item = Main.rand.Next(list);
+			list.Remove(item);
```

Comment:
I don't love that you have to make a new list every time to do this, but I don't have a better idea